07_errors/patrickmarabeas/errors_test.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnknownError(t *testing.T) {
+	a := assert.New(t)
+
+	error := NewError(123)
+	a.Equal(error, NewError(Unknown))
+}
+
+func TestError(t *testing.T) {
+	a := assert.New(t)
+
+	error := Error{1, "message"}
+	formattedError := error.Error()
+	a.Equal(formattedError, "[1] message")
+}
```

Comment:
I don't see that you have 100% test coverage with this commit. There are a couple of branches in `NewError()` that are not tested. They probably get tested in a later commit by using it, but since you have structured your PR as a series of small cohesive commits (good), you should ensure each commit passes all the tests and has complete coverage. There is some `git rebase --exec` magic you can use to run all your tests and coverage checks on every one of your commits automagically, but that usually requires a makefile or script to run all the checks you want to.

I will share the magic in a subsequent comment where I have noticed your tests failing...